econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -0,0 +1,1056 @@
+                               get_inference=get_inference,
+                               props=props,
+                               expand_arr=expand_arr,
+                               drop_sample=False)  # dropping the sample dimension is handled above instead
+
+    def _dict_summary(self, get_inference, *, props, kind, drop_sample=False, expand_arr=False):
+        """
+        Summarizes results into a dictionary.
+
+        Parameters
+        ----------
+        get_inference : lambda
+            Method to get the relevant inference results from each result object
+        props : list of (string, string or lambda)
+            Set of column names and ways to get the corresponding values from the inference object
+        kind : string
+            The kind of inference results to get (e.g. 'global', 'local', or 'cohort')
+        drop_sample : boolean, default False
+            Whether to drop the sample dimension from each array
+        expand_arr : boolean, default False
+            Whether to add an initial sample dimension to the result arrays
+        """
+        def make_dict(props):
+            # should be serialization-ready and contain no numpy arrays
+            res = _get_default_specific_insights(kind)
+            res.update([(key, value.tolist()) for key, value in props])
+            return {**self._shared, **res}
+
+        return self._summarize(summary=make_dict,
+                               get_inference=get_inference,
+                               props=props,
+                               expand_arr=expand_arr,
+                               drop_sample=drop_sample)
+
+    def global_causal_effect(self, alpha=0.05):
+        """
+        Get the global causal effect for each feature as a pandas DataFrame.
+
+        Parameters
+        ----------
+        alpha : the confidence level of the confidence interval
+
+        Returns
+        -------
+        global_effects : pandas Dataframe
+            DataFrame with the following structure:
+
+            :Columns: ['point', 'stderr', 'zstat', 'pvalue', 'ci_lower', 'ci_upper']
+            :Index: ['feature', 'feature_value']
+            :Rows: For each feature that is numerical, we have an entry with index ['{feature_name}', 'num'], where
+                    'num' is literally the string 'num' and feature_name is the input feature name.
+                    For each feature that is categorical, we have an entry with index ['{feature_name}',
+                    '{cat}v{base}'] where cat is the category value and base is the category used as baseline.
+                    If all features are numerical then the feature_value index is dropped in the dataframe, but not
+                    in the serialized dict.
+        """
+        # a global inference indicates the effect of that one feature on the outcome
+        return self._pandas_summary(lambda res: res.global_inference, props=self._point_props(alpha), n=1)
```

Comment:
This requires some change if global_inference results is either a pop summary or a point estimate summary dependent on the heterogeneity model